nui-libgdx/src/main/java/org/terasology/nui/backends/libgdx/LibGDXFont.java

```diff
@@ -44,31 +46,31 @@ public GlyphLayout getGlyphLayout() {
     public int getWidth(String text) {
         glyphLayout.reset();
         glyphLayout.setText(bitmapFont, text);
-        return (int) glyphLayout.width;
+        return Math.roundUsing(glyphLayout.width, RoundingMode.CEILING);
     }
 
     @Override
     public int getWidth(Character c) {
         glyphLayout.reset();
         glyphLayout.setText(bitmapFont, "" + c);
-        return (int) glyphLayout.width;
+        return Math.roundUsing(glyphLayout.width, RoundingMode.CEILING);
     }
 
     @Override
     public int getHeight(String text) {
         glyphLayout.reset();
         glyphLayout.setText(bitmapFont, text);
-        return (int) glyphLayout.height;
+        return Math.roundUsing(glyphLayout.height, RoundingMode.CEILING);
     }
 
     @Override
     public int getLineHeight() {
-        return (int) bitmapFont.getLineHeight();
+        return Math.roundUsing(bitmapFont.getLineHeight(), RoundingMode.CEILING);
     }
 
     @Override
     public int getBaseHeight() {
-        return (int) bitmapFont.getCapHeight();
+        return Math.roundUsing(bitmapFont.getCapHeight(), RoundingMode.CEILING);
```

Comment:
not familiar with the details of what is considered "cap", "base", or "line" height here, but I'm wondering whether something called "cap height" should be rounded up... I would understand it to be an upper cap, so rounding up would exceed the cap? :thinking: just thinking out loud here...

Reply:
According to [the documentation](https://javadoc.io/static/com.badlogicgames.gdx/gdx/1.9.14/com/badlogic/gdx/graphics/g2d/BitmapFont.html#getCapHeight--), the cap is the maximum height of most capital letters from the baseline. So, it's not the absolute maximum, just the average. The absolute maximum is the `capHeight + ascent`. Rounding-up will likely exceed the cap a bit, however that's the only way I know of to guarentee that we don't under-fit. If the cap's already an integer value, then it shouldn't change. I'm open to ideas for improvement here though.

The main problem with under-fitting is that it leads to unexpected line-wrapping, since the calculated layout size is different to the actual size. Usually this leads to text lines being seen as fitting into a smaller space than they actually would. See https://github.com/MovingBlocks/DestinationSol/pull/649#pullrequestreview-887883260 for an visual example of this issue happening (it usually manifests itself as word wrapping one letter early).

Reply:
Thanks for the explanation, I think this approach is fine as an improvement / fix for the problem you're experiencing.